cms/core/migrations/0006_update_user_groups.py

```diff
@@ -0,0 +1,192 @@
+
+    for model, app in snippet_classes_dict.items():
+        for permission in WAGTAIL_PERMISSION_TYPES:
+            assign_permission_to_group(
+                apps,
+                settings.PUBLISHING_ADMINS_GROUP_NAME,
+                permission_codename=f"{permission}_{model.lower()}",
+                app=app,
+                model=model,
+            )
+        if model in ("MainMenu", "FooterMenu"):
+            # MainMenu and FooterMenu are publishable
+            assign_permission_to_group(
+                apps,
+                settings.PUBLISHING_ADMINS_GROUP_NAME,
+                permission_codename=f"publish_{model.lower()}",
+                app=app,
+                model=model,
+            )
+
+
+def create_bundle_permissions(apps):
+    """Grant all permissions on Bundles to Publishing Admins and Publishing Officers,
+    and view-only access to Viewers.
+    """
+    bundles_app = "bundles"
+    bundle_class = "Bundle"
+
+    for permission in WAGTAIL_PERMISSION_TYPES:
+        for group_name in [settings.PUBLISHING_ADMINS_GROUP_NAME, settings.PUBLISHING_OFFICERS_GROUP_NAME]:
+            assign_permission_to_group(
+                apps,
+                group_name=group_name,
+                permission_codename=f"{permission}_{bundle_class.lower()}",
+                app=bundles_app,
+                model=bundle_class,
+            )
+
+    # View-only access for the Viewers group
+    assign_permission_to_group(apps, settings.VIEWERS_GROUP_NAME, "view_bundle", bundles_app, bundle_class)
+
+
+def create_page_permissions(apps):
+    """Allow Publishing Admins and Publishing Officers all permissions on the HomePage and its children."""
+    # TODO: these shouldn't be imported directly and should use historical models instead
+    from django.contrib.auth.models import Group
+
+    from cms.home.models import HomePage
+
+    # HomePage = apps.get_model("home.HomePage")
+    # Group = apps.get_model("auth.Group")
+
+    home_page = HomePage.objects.first()
+
+    for group_name in [settings.PUBLISHING_ADMINS_GROUP_NAME, settings.PUBLISHING_OFFICERS_GROUP_NAME]:
+        group = Group.objects.get(name=group_name)
+        for permission_type in WAGTAIL_PAGE_PERMISSION_TYPES:
+            if group_name == settings.PUBLISHING_OFFICERS_GROUP_NAME and permission_type not in ("add", "change"):
+                continue  # Only allow Publishing Officers to add and change pages
+            GroupPagePermission.objects.create(group=group, page=home_page, permission_type=permission_type)
+
+
+def create_topic_page_featured_series_permission(apps):
+    """Create a custom permission that will be registered using a Wagtail hook,
+    given to the Publishing Admins group,
+    and used on the TopicPage's featured article series FieldPanel.
+    """
+    assign_permission_to_group(
+        apps,
+        group_name=settings.PUBLISHING_ADMINS_GROUP_NAME,
+        permission_codename="add_featured_article_series_on_topic_page",
```

Comment:
question: where does this requirement come from? The RBAC matrix mentions "headline figures", not "highlighted/featured series"

Reply:
From [CMS-62 Topic page](https://jira.ons.gov.uk/browse/CMS-62):

> Automation logic:
> Headline figures - if no headline figures are added editorially, but there is a featured bulletin series, attempt to pull the headline figures from the latest bulletin in the featured series

So apart from the manual headline figure selection as proposed in [CMS-349 Headline figures spike](https://jira.ons.gov.uk/browse/CMS-349), if we want to only allow the Publishing Admins to set the headline figures, we'll need to restrict who can add the Featured series

Reply:
Had a chat with Dan and removed in [3dbd513](https://github.com/ONSdigital/dis-wagtail/pull/128/commits/3dbd513619abdce3feb87767ae19487560ad0bf0)